### PR TITLE
vmware fusion headless

### PR DESCRIPTION
**Pull request by davidino**

### CODE REVIEW

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+```
+
+As you can see from the script, you have to do one more thing: define the `env` variable `vmFile`, (put it in your .bash_profile ( or .zshrc as well ) in that way:
+
+```
+export vmFile="/Users/username/Documents/Virtual Machines.localized/Ubuntu 15.04 server64-bit.vmwarevm/Ubuntu 15.04 server64-bit.vmx"
+```
+
+Ok, that's all. Now you can handle you vmware vm from the shell. We are using it on osx, but should be even on windows systems.
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: /should be even on windows systems/should work on windows too/
(the helper script might work with [cygwin](https://www.cygwin.com/))

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+```
+$ vmrun start
+$ vmrun stop
+$ vmrun pause
+...
+
+```
+
+As you can see from the script, you have to do one more thing: define the `env` variable `vmFile`, (put it in your .bash_profile ( or .zshrc as well ) in that way:
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: / you have to do/you need/

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    ;;
+  status)
+    vmrun list
+    ;;
+  *)
+    echo "Usage: possible options are: start | stop | suspend | pause | unpause | reset | status"
+    exit 1
+esac
+```
+Once you had this file in the path `/usr/local/bin/vm` (or the path you prefer) you can run the commands like:
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: `/usr/local/bin/vmrun`

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+```
+$ vmrun start
+$ vmrun stop
+$ vmrun pause
+...
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: /vmrun/vm/g

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+<!-- more -->
+
+The command is `vmrun` and should be located on `/Applications/VMwareFusion.app/Contents/Library/vmrun`. The `power-commands` are all the actions that you can send to the virtual machine (`start|stop|pause|halt`). You can get all the command running the `vmrun --help`. If you want to know more about this command you can read the man file.
+
+You have also to specify the virtual machine vmx file that you want to handle. By default Vmware puts all the images in the Document user folder.
+
+In order to use the `vmrun` command without all the path, let's create a symbolic link:
+
+`sudo ln -s "/Applications/VMware Fusion.app/Contents/Library/vmrun" /usr/bin/vmrun`
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: vmware's manual says on windows it should be in `c:\Program Files\VMware\VMware Workstation`
windows's path are so odd, beside they use the wrong slash, am I the only creeped out by those `<letter>:` mounting points? :P

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 6]
```diff
@@ -0,0 +1,81 @@
+---
+layout: post
+title: "VMware fusion headless"
+date: 2015-07-27 18:23
+comments: true
+categories: osx, vmware fusion, vm
+author: David Funaro,
+---
+Nobody is talking about that. But there is, and it works fine. As `Virtualbox` can do, even `Vmware Fusion` is able to handle a virtual machine in headless mode (writing commands from shell). In that way you don't need to keep the vmware window on top of the desktop, and even you can automate stuff, if you need.
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: /As `Virtualbox` can do, even `Vmware Fusion` is able to handle a virtual machine in headless mode (writing commands from shell)/Just like  `Virtualbox`, `Vmware Fusion` is able to handle a virtual machine in headless mode too (with a little bit of shell fu)/

File: source/_posts/2015-07-27-vmware-fusion-headless.markdown [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+Searching on the web, we finally found a command to do that:
+
+```
+vmrun power-commands <path-to-vmx-file> nogui
+```
+
+<!-- more -->
+
+The command is `vmrun` and should be located on `/Applications/VMwareFusion.app/Contents/Library/vmrun`. The `power-commands` are all the actions that you can send to the virtual machine (`start|stop|pause|halt`). You can get all the command running the `vmrun --help`. If you want to know more about this command you can read the man file.
```
Comment (unlucio) [inline source/_posts/2015-07-27-vmware-fusion-headless.markdown]: /all the command/all the possible options || the commands list/

**Comment by unlucio** [comment]
:+1:

**Comment by davidino** [comment]
@unlucio thanks!
@cirpo tell me if we can publish it.

**Comment by cirpo** [comment]
@davidino sure, lets check it together right after lunch, ok?

**Comment by davidino** [comment]
:+1:

**Comment by davidino** [comment]
@cirpo check now :)

**Comment by davidino** [comment]
@geshan / @razanbilwani can you please check ?

Thanks

**Comment by davidino** [comment]
thanks @razanbilwani. 
@cirpo we are ready :)

**Comment by razanbilwani** [comment]
i love this post. will try on windows!

**Comment by cirpo** [comment]
@davidino lets go live then :)

**Comment by davidino** [comment]
[it's live](http://tech.namshi.com/blog/2015/07/29/vmware-fusion-headless/)

**Comment by shidhincr** [comment]
:+1:

**Comment by geshan** [comment]
I think after going live with this post newsletter 101 is missing, may be its not re-based with source.

**Comment by davidino** [comment]
fixed, thanks @geshan !

**Pull request merged**